integration-cli/docker_cli_events_test.go

```diff
@@ -203,68 +203,40 @@ func TestEventsImageImport(t *testing.T) {
-	// function as a die event (some sort of race)
-	// I am not proud of this - jessfraz
-	d := NewDaemon(t)
-	if err := d.StartWithBusybox(); err != nil {
-		t.Fatalf("Could not start daemon with busybox: %v", err)
-	}
-	defer d.Stop()
-
 	since := time.Now().Unix()
-	out, err := d.Cmd("run", "--rm", "busybox", "true")
+	out, _, err := runCommandWithOutput(exec.Command(dockerBinary, "run", "--rm", "busybox", "true"))
 	if err != nil {
 		t.Fatal(out, err)
 	}
-	out, err = d.Cmd("run", "--rm", "busybox", "true")
+	out, _, err = runCommandWithOutput(exec.Command(dockerBinary, "run", "--rm", "busybox", "true"))
 	if err != nil {
 		t.Fatal(out, err)
 	}
-	out, err = d.Cmd("events", fmt.Sprintf("--since=%d", since), fmt.Sprintf("--until=%d", time.Now().Unix()), "--filter", "event=die")
+	out, _, err = runCommandWithOutput(exec.Command(dockerBinary, "events", fmt.Sprintf("--since=%d", since), fmt.Sprintf("--until=%d", time.Now().Unix()), "--filter", "event=die"))
 	if err != nil {
 		t.Fatalf("Failed to get events: %s", err)
 	}
 	events := strings.Split(out, "\n")
 	events = events[:len(events)-1]
-	if len(events) != 2 {
-		t.Fatalf("Expected 2 events, got %d: %v", len(events), events)
-	}
-	dieEvent := strings.Fields(events[len(events)-1])
-	if dieEvent[len(dieEvent)-1] != "die" {
-		t.Fatalf("event should be die, not %#v", dieEvent)
-	}
-
-	dieEvent = strings.Fields(events[len(events)-2])
-	if dieEvent[len(dieEvent)-1] != "die" {
-		t.Fatalf("event should be die, not %#v", dieEvent)
+	for _, event := range events {
+		eventFields := strings.Fields(event)
+		if eventFields[len(eventFields)-1] != "die" {
+			t.Fatalf("event should be die, not %#v", eventFields)
+		}
 	}
 
-	out, err = d.Cmd("events", fmt.Sprintf("--since=%d", since), fmt.Sprintf("--until=%d", time.Now().Unix()), "--filter", "event=die", "--filter", "event=start")
+	out, _, err = runCommandWithOutput(exec.Command(dockerBinary, "events", fmt.Sprintf("--since=%d", since), fmt.Sprintf("--until=%d", time.Now().Unix()), "--filter", "event=die", "--filter", "event=start"))
 	if err != nil {
 		t.Fatalf("Failed to get events: %s", err)
 	}
 	events = strings.Split(out, "\n")
 	events = events[:len(events)-1]
-	if len(events) != 4 {
-		t.Fatalf("Expected 4 events, got %d: %v", len(events), events)
-	}
-	startEvent := strings.Fields(events[len(events)-4])
-	if startEvent[len(startEvent)-1] != "start" {
-		t.Fatalf("event should be start, not %#v", startEvent)
-	}
-	dieEvent = strings.Fields(events[len(events)-3])
-	if dieEvent[len(dieEvent)-1] != "die" {
-		t.Fatalf("event should be die, not %#v", dieEvent)
-	}
-	startEvent = strings.Fields(events[len(events)-2])
-	if startEvent[len(startEvent)-1] != "start" {
-		t.Fatalf("event should be start, not %#v", startEvent)
-	}
-	dieEvent = strings.Fields(events[len(events)-1])
-	if dieEvent[len(dieEvent)-1] != "die" {
-		t.Fatalf("event should be die, not %#v", dieEvent)
+	for _, event := range events {
```

Comment:
you and @lk4d4 keeping things spec like :D

On Tue, Feb 24, 2015 at 4:44 PM, Jessica Frazelle <jess@docker.com> wrote:

> ahhh ok ok
>
> On Tue, Feb 24, 2015 at 4:43 PM, Ahmet Alp Balkan <
> notifications@github.com> wrote:
>
>> In integration-cli/docker_cli_events_test.go
>> <https://github.com/docker/docker/pull/10998#discussion_r25308862>:
>>
>> > -	startEvent := strings.Fields(events[len(events)-4])
>> > -	if startEvent[len(startEvent)-1] != "start" {
>> > -		t.Fatalf("event should be start, not %#v", startEvent)
>> > -	}
>> > -	dieEvent = strings.Fields(events[len(events)-3])
>> > -	if dieEvent[len(dieEvent)-1] != "die" {
>> > -		t.Fatalf("event should be die, not %#v", dieEvent)
>> > -	}
>> > -	startEvent = strings.Fields(events[len(events)-2])
>> > -	if startEvent[len(startEvent)-1] != "start" {
>> > -		t.Fatalf("event should be start, not %#v", startEvent)
>> > -	}
>> > -	dieEvent = strings.Fields(events[len(events)-1])
>> > -	if dieEvent[len(dieEvent)-1] != "die" {
>> > -		t.Fatalf("event should be die, not %#v", dieEvent)
>> > +	for _, event := range events {
>>
>> @jfrazelle <https://github.com/jfrazelle> inline anonymous go functions?
>> :)
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/docker/docker/pull/10998/files#r25308862>.
>>
>
>